BraintreeCore/src/main/java/com/braintreepayments/api/PaymentMethodNonce.kt

```diff
@@ -0,0 +1,47 @@
+package com.braintreepayments.api
+
+import android.os.Parcelable
+import android.os.Parcel
+import com.braintreepayments.api.PaymentMethodNonce
+
+/**
+ * Base class representing a method of payment for a customer. [PaymentMethodNonce] represents the
+ * common interface of all payment method nonces, and can be handled by a server interchangeably.
+ *
+ * @property string The nonce generated for this payment method by the Braintree gateway. The nonce will
+ * represent this PaymentMethod for the purposes of creating transactions and other monetary
+ * actions.
+ * @property isDefault `true` if this payment method is the default for the current customer, `false` otherwise
+ */
+open class PaymentMethodNonce : Parcelable {
+
+    companion object {
+        @JvmField
+        val CREATOR = object : Parcelable.Creator<PaymentMethodNonce> {
+            override fun createFromParcel(`in`: Parcel) = PaymentMethodNonce(`in`)
+            override fun newArray(size: Int) = arrayOfNulls<PaymentMethodNonce>(size)
+        }
+    }
+
+    open val string: String
+    open val isDefault: Boolean
+
+    protected constructor(nonce: String, isDefault: Boolean) {
+        string = nonce
+        this.isDefault = isDefault
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    override fun writeToParcel(dest: Parcel, flags: Int) {
+        dest.writeString(string)
+        dest.writeByte(if (isDefault) 1.toByte() else 0.toByte())
+    }
+
+    protected constructor(`in`: Parcel) {
+        string = `in`.readString()!!
```

Comment:
Can you do this similar to nil-coalescing in Swift? => 'in'.readString() ?: ""

Reply:
Technically yes, but `Parcelable` it's weird. I'd leave it only because the above `writeToParcel()` method shows that the `string` property will always be present when the data is deserialized.

We will eventually delete this code at some point in favor of [@Parcelize](https://developer.android.com/kotlin/parcelize).